### ISSUE TITLE
App crash on startup after upgrade to ver 3.0.1

### ISSUE THREAD

**Issue by instantlinux** [nextcloud/ios#1361]
### Expected behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             

I should be able to update to a new version without fear of losing use of the software (see also #766, #850--it's terrifying for us users). The app should come up and allow me to connect and use its UI features.

### Actual behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               

App crashes after displaying the logo splash page for less than 1 second.

### Steps to reproduce     

1. Set up the same version of iOS that I have on my iPhone (as below)
2. Install the previous version of NextCloud app (2.25.9)
3. Perform the upgrade to the app version noted below
4. Try to run the app

### Reasoning or why should it be changed/implemented?

Apple provides no viable way to revert an iOS app update so I'm stuck without being able to use nextcloud, period.

### iOS version

12.1.4

### App version

3.0.1

### Server configuration

18.0.7 docker image posted at hub.docker.com

**Operating system:**

Debian 10 buster

**Web server:**

Apache2 2.4.38

**Database:**

Mariadb 10.4.13

**PHP version:**

7.3.20

**Nextcloud version:** (see Nextcloud admin page)

18.0.7

**Comment by instantlinux** [comment]
To recover from this, I had to delete the app and reinstall (by now, 3.0.2 is out). Evidently, something leftover on the phone from the previous version was preventing startup. After reinstall, I ran into the issue cited in #1365. Had to:

* Try to recall what settings I'd made in this app's settings (auto-upload is disabled by default, so it has to be enabled and configured with whatever settings I used to have), got it wrong at first, finally got it right
* Use a DIFFERENT app to copy the photos that had been taken in the two weeks since this failure happened
* Choose a "legacy" login method to reconnect the app to nextcloud server

Hopefully the experience can be improved for seamless future upgrades, or if not, notes on upgrades can be added to release notes of this repo so we can get up and running without these headaches.
